lib/elixir/src/elixir_module.erl

```diff
@@ -121,6 +123,24 @@ compile(Line, Module, Block, Vars, E) ->
     elixir_code_server:call({undefmodule, Ref})
   end.
 
+validate_compile_opts([], _Defs, _Unreachable, _File) ->
+  [];
+validate_compile_opts([Opts | Rest], Defs, Unreachable, File) when is_list(Opts) ->
+  validate_compile_opts(Opts, Defs, Unreachable, File) ++ validate_compile_opts(Rest, Defs, Unreachable, File);
+validate_compile_opts([{inline, Inlines} | Rest], Defs, Unreachable, File) ->
+  case Inlines -- Defs of
+    [] ->
+      [{inline, Inlines -- Unreachable} | validate_compile_opts(Rest, Defs, Unreachable, File)];
+    [{Name, Arity} | _] ->
+      elixir_errors:form_error([], File, ?MODULE, {bad_inline, Name, Arity})
+  end;
+%% TODO: Make this an error and skip parse transform processing on 2.0.
+validate_compile_opts([{parse_transform, Module} = Opt | Rest], Defs, Unreachable, File) ->
+  elixir_errors:form_warn([], File, ?MODULE, {parse_transform, Module}),
```

Comment:
Let's include the line as first argument: `[{line, Line}]`. You will need to pass the `Line` all the way in.